gluten-substrait/src/main/scala/org/apache/gluten/planner/cost/RoughCostModel.scala

```diff
@@ -34,6 +36,10 @@ class RoughCostModel extends LongCostModel {
         // Make trivial ProjectExec has the same cost as ProjectExecTransform to reduce unnecessary
         // c2r and r2c.
         10L
+      case r2c: RowToColumnarExecBase if hasComplexTypes(r2c.schema) =>
+        // Avoid moving computation back to native when transition has complex types in schema.
+        // Such transitions are observed to be extreme expensive as of now.
+        Long.MaxValue
```

Comment:
Can we add a more reasonable value to the cost? How can we measure the operator cost now?

Reply:
We can use it as a temp solution. The r2c is very expensive.

Reply:
Yes. It's just a heuristic as a temp solution.